src/RegistrationController.js

```diff
@@ -106,7 +106,7 @@ function (
         });
       } else {
         // register via activation email
-        this.model.appState.set('username', this.model.get('userName'));
+        this.model.appState.set('username', this.model.get('email'));
```

Comment:
I saw this in the description `New self service registration logic does not use userName attribute. Instead it uses email and optionally login attributes`. Do we need to handle login attribute as well ? Will email always be available ?

Reply:
when email as login feature is enabled, only email is available. Therefore email looks like a common attribute that we can use irrespective of email as login feature